lib/ansible/module_utils/alicloud_ecs.py

```diff
@@ -0,0 +1,193 @@
+# This particular file snippet, and this file snippet only, is BSD licensed.
+# Modules you write using this snippet, which is embedded dynamically by Ansible
+# still belong to the author of the module, and may assign their own license
+# to the complete work.
+#
+# Copyright (c) 2017 Alibaba Group Holding Limited. He Guimin <heguimin36@163.com.com>
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright notice,
+#      this list of conditions and the following disclaimer in the documentation
+#      and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+import os
+
+try:
+    import footmark
+    import footmark.ecs
+    import footmark.slb
+    import footmark.vpc
+    import footmark.rds
+    HAS_FOOTMARK = True
+except ImportError:
+    HAS_FOOTMARK = False
+
+
+class AnsibleACSError(Exception):
+    pass
+
+
+def acs_common_argument_spec():
+    return dict(
+        alicloud_access_key=dict(aliases=['acs_access_key', 'ecs_access_key', 'access_key']),
+        alicloud_secret_key=dict(no_log=True, aliases=['acs_secret_access_key', 'ecs_secret_key', 'secret_key']),
+        alicloud_security_token=dict(aliases=['security_token', 'access_token'], no_log=True),
+    )
+
+
+def ecs_argument_spec():
+    spec = acs_common_argument_spec()
+    spec.update(
+        dict(
+            alicloud_region=dict(aliases=['acs_region', 'ecs_region', 'region']),
+        )
+    )
+    return spec
+
+
+def get_acs_connection_info(module):
+
+    # Check module args for credentials, then check environment vars access_key and region
+
+    access_key = module.params.get('alicloud_access_key')
+    secret_key = module.params.get('alicloud_secret_key')
+    security_token = module.params.get('alicloud_security_token')
+    region = module.params.get('alicloud_region')
+
+    if not access_key:
+        if 'ALICLOUD_ACCESS_KEY' in os.environ:
```

Comment:
You can add this directly into `acs_common_argument_spec`

Eg
`'username': dict(fallback=(env_fallback, ['ANSIBLE_NET_USERNAME'])),`